src/partition/ReceivedPartition.cpp

```diff
@@ -450,15 +454,15 @@ void ReceivedPartition::prepareBoundingBox()
 
   PRECICE_DEBUG("Merge bounding boxes and increase by safety factor");
 
-  // Create BB around both "other" meshes
-  if (_fromMapping) {
-    auto other_bb = _fromMapping->getOutputMesh()->getBoundingBox();
+  // Create BB around all "other" meshes
+  for (mapping::PtrMapping fromMapping : _fromMappings) {
```

Comment:
We should take the shared pointer by reference to prevent copies.
```suggestion
  for (mapping::PtrMapping& fromMapping : _fromMappings) {
```